x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/dataframe/RestStartDataFrameAnalyticsAction.java

```diff
@@ -3,36 +3,38 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.xpack.ml.rest.analytics;
+package org.elasticsearch.xpack.ml.rest.dataframe;
 
 import org.elasticsearch.client.node.NodeClient;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.rest.BaseRestHandler;
 import org.elasticsearch.rest.RestController;
 import org.elasticsearch.rest.RestRequest;
 import org.elasticsearch.rest.action.RestToXContentListener;
-import org.elasticsearch.xpack.core.ml.action.RunAnalyticsAction;
+import org.elasticsearch.xpack.core.ml.action.StartDataFrameAnalyticsAction;
+import org.elasticsearch.xpack.core.ml.dataframe.DataFrameAnalyticsConfig;
 import org.elasticsearch.xpack.ml.MachineLearning;
 
 import java.io.IOException;
 
-public class RestRunAnalyticsAction extends BaseRestHandler {
+public class RestStartDataFrameAnalyticsAction extends BaseRestHandler {
 
-    public RestRunAnalyticsAction(Settings settings, RestController controller) {
+    public RestStartDataFrameAnalyticsAction(Settings settings, RestController controller) {
         super(settings);
-        controller.registerHandler(RestRequest.Method.POST, MachineLearning.BASE_PATH + "analytics/{index}/_run", this);
+        controller.registerHandler(RestRequest.Method.POST, MachineLearning.BASE_PATH + "data_frame/analytics/{"
+            + DataFrameAnalyticsConfig.ID.getPreferredName() + "}/_start", this);
```

Comment:
this API does not allow in body id's, I think we should either support it for all endpoints possible or not

Reply:
Good spot. Will add.

Reply:
Actually, not so sure about this. As this currently only expects the `id` which has to be part of the URL, it doesn't make sense. It seems like the same is done for starting the data frame job.